### ISSUE TITLE
Support encrypted rclone configurations

### ISSUE THREAD

**Issue by bytesnz** [restic/restic#3672]
Output of `restic version`
--------------------------
`restic 0.12.1 compiled with go1.16.6 on linux/amd64`

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------

Add support for [rclone configuration encryption](https://rclone.org/docs/#configuration-encryption). Currently, it logs rclone asking for the password, but you can't enter it
```
restic -r rclone:pcloud:backups --password-file secret forget --keep-daily 3 --keep-weekly 5 --keep-monthly 12
rclone: Enter configuration password:
rclone: password:2022/03/18 15:32:24 ERROR : Couldn't decrypt configuration, most likely wrong password.
rclone: Enter configuration password:
```

What are you trying to do? What problem would this solve?
---------------------------------------------------------
Use an encrypted rclone configuration. Currently have to unencrypt the configuration before running restic.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
It always does, would just be nice not to have to unencrypt, the reencrypt the rclone config

**Comment by DataHearth** [comment]
Hey! Any update on that ? I have the same problem. I'm not a big fan of having configurations credentials exposed on my system :)

**Comment by MichaelEischer** [comment]
I'm not sure whether it is possible to implement this on the restic side. stdin and stdout are required for the communication between restic and rclone. So rclone would have to read the password directly from the terminal.

**Comment by DataHearth** [comment]
Maybe it can be implemented by storing encrypted credentials on restic side, then decrypt it when needed and sorta sending it to `stdin` using `os.Stdin`. I haven't looked at how restic and rclone are made, but if it "only" involved `stdout` and `stdin`, using the `os` module should do the trick, right ?

**Comment by MichaelEischer** [comment]
restic has no way of knowing whether rclone will ask for a password or not. So we can't just blindly send something along.

**Comment by bytesnz** [comment]
Could you check for a password prompt on the rclone stdin?

restic could detect if the config is encrypted by checking the rclone config file (see encrypted config head below), but it is quite nasty. 
```
# Encrypted rclone configuration File

RCLONE_ENCRYPT_V0:
...
```

Maybe @ncw or someone from rclone has some ideas?

**Comment by MichaelEischer** [comment]
The cleanest solution would be for rclone to read the password directly from the terminal. There's no point in restic trying to somehow read the rclone config, that will just cause an endless series of problems.

Checking for a password prompt from rclone will also be total mess as we would have to capture stderr, and parse it which seems like a horrible idea.

**Comment by ncw** [comment]
Check out the [--password-command](https://rclone.org/docs/#password-command-spaceseplist) flag (which like all rclone flags is also an environment variable.

That lets you run a program to read the password - you can use that to read the config password from your keychain quite easily.

**Comment by ivlis** [comment]
```
restic - rclone.args="serve restic --stdio --verbose --config /rclone.rc --password-command 'cat ${RCLONE_PASSWORD_FILE}' "
```

**Comment by MichaelEischer** [comment]
Closing as the issue appears to be solved.

**Comment by m0byn** [comment]
Previously, I used restic with rclone to backup some files to Proton Drive. Then, I decided that it is a good idea to encrypt the config file of rclone. As a result, the previously working restic command does not work anymore. See the command here: 
`restic backup /home/manuel/ --one-file-system --repo rclone:PD:ResticBackup`

Unfortunately, I am not that familiar with rclone and restic. Could maybe @ivlis or @ncw help me figuring out how to properly adapt the previously used command with using an encrypted rclone config?
